### PR TITLE
PDA-47 Process Diagram Endpoint

### PR DESCRIPTION
**Pull request by ffleandro**

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by werbth** [comment]
retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by ffleandro** [comment]
Retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by ffleandro** [comment]
Retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

File: src/main/java/org/entando/plugins/pda/pam/service/process/model/KieProcessId.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    private String containerId;
+    private Long processId;
+
+    public KieProcessId(String id) {
+        String[] split = Optional.ofNullable(id)
+                .orElseThrow(() -> new BadRequestException("org.entando.kie.error.id"))
+                .split(SEPARATOR);
+
+        if (split.length != 2) { //NOPMD
```
Comment (werbth) [inline src/main/java/org/entando/plugins/pda/pam/service/process/model/KieProcessId.java]: Create a constant for this literal and you can remove that `NOPMD` comment.

File: src/main/java/org/entando/plugins/pda/pam/service/group/KieGroupService.java [hunk 2]
```diff
@@ -8,31 +8,31 @@
...
-            String[] inGroups = associatedEntityDefinitions.getAssociatedEntities().values().stream()
-                    .flatMap(Arrays::stream).toArray(String[]::new);
-            return inGroups.length == 0 ? Collections.emptyList() : customQueryService.getGroups(connection, inGroups);
+    public List<String> list(Connection connection, String id) {
+        if (id == null) {
+            return customQueryService.getGroups(connection);
         }
-        return customQueryService.getGroups(connection);
+
+        KieProcessId compositeId = new KieProcessId(id);
```
Comment (werbth) [inline src/main/java/org/entando/plugins/pda/pam/service/group/KieGroupService.java]: The processId here is the process definition. It is a string, so it cannot be used with this KieProcessId object.

File: src/test/java/org/entando/plugins/pda/pam/service/group/KieGroupServiceTest.java [hunk 3]
```diff
@@ -53,7 +54,7 @@ public void shouldListGroupsFromProcessId() {
         // Given
         ProcessServicesClient processServicesClient = mock(ProcessServicesClient.class);
         String containerId = "container-1";
-        String processId = "process-1";
+        String processId = "1";
```
Comment (werbth) [inline src/test/java/org/entando/plugins/pda/pam/service/group/KieGroupServiceTest.java]: This should be a string with alphanumeric values.

**Review by werbth — CHANGES_REQUESTED** [review]

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by werbth** [comment]
Retest this please

**Comment by entando-service** [comment]
All is well

**Review by werbth — APPROVED** [review]

**Pull request merged**